clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt

```diff
@@ -44,12 +60,61 @@ interface FossIdRestService {
         /**
          * The mapper for JSON (de-)serialization used by this service.
          */
-        val JSON_MAPPER = JsonMapper()
+        val JSON_MAPPER: ObjectMapper = JsonMapper()
             .setPropertyNamingStrategy(PropertyNamingStrategies.SNAKE_CASE)
             // FossID has a bug in get_results/scan.
             // Sometimes the match_type is "ignored", sometimes it is "Ignored".
-            .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS)
-            .registerKotlinModule()
+            .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS).copy()
+            .registerModule(
+                kotlinModule()
+                    .setDeserializerModifier(FossIdDeserializerModifier())
+            )
+
+        /**
+         * A class to modify the standard Jackson deserialization to deal with inconsistencies in responses
+         * sent by the FossID server.
+         * FossID usually returns a List or a Map, but if there is no entry, it returns data:false.
+         * To handle this special case, this class makes sure that a custom deserializer is used for
+         * lists or maps that detects such a result and converts it to an empty list or map, respective.
+         * That way, the FossID service interface can use meaningful result types.
+         */
+        private class FossIdDeserializerModifier : BeanDeserializerModifier() {
+            override fun modifyMapDeserializer(
+                config: DeserializationConfig,
+                type: MapType,
+                beanDesc: BeanDescription,
+                deserializer: JsonDeserializer<*>
+            ): JsonDeserializer<*> {
+                return LenientDeserializer(deserializer) { mutableMapOf<Any, Any>() }
+            }
+
+            override fun modifyCollectionDeserializer(
+                config: DeserializationConfig?,
+                type: CollectionType?,
+                beanDesc: BeanDescription?,
+                deserializer: JsonDeserializer<*>
+            ): JsonDeserializer<*> {
+                return LenientDeserializer(deserializer) { mutableListOf<Any>() }
+            }
+        }
+
+        /** Custom Jackson deserializer that abstracts the behaviour explained above.
+         * [creator] is a function that creates an empty object when data:false is received.
+         * Otherwise, the standard Kotlin/Jackson deserializer is called by delegation.
+         */
+        private class LenientDeserializer(delegate: JsonDeserializer<*>, val creator: () -> Any) :
```

Comment:
The name `LenientDeserializer` doesn't really explain in what way the deserializer is "lenient". I propose something more telling like e.g. "FalseValueDeserializer".

Reply:
`val creator` could be private.